lib/LaTeXML/Core/Parameter.pm

```diff
@@ -113,17 +113,8 @@ sub reparse {
   elsif ($$self{semiverbatim}) {                            # Needs neutralization
     return $tokens->neutralize(@{ $$self{semiverbatim} }); }    # but maybe specific to catcodes
   else {
-    return $gullet->readingFromMouth(LaTeXML::Core::Mouth->new(), sub {    # start with empty mouth
-        my ($gulletx) = @_;
-        my @tokens = $tokens->unlist;
-        if (@tokens    # Strip outer braces from dimensions & friends
-          && ($$self{type} =~ /^(?:Number|Dimension|Glue|MuDimension|MuGlue)$/)
-          && ($tokens[0]->getCatcode == CC_BEGIN) && ($tokens[-1]->getCatcode == CC_END)) {
-          shift(@tokens); pop(@tokens); }
-        $gulletx->unread(@tokens);    # but put back tokens to be read
-        my $value = $self->read($gulletx);
-        $gulletx->skipSpaces;
-        return $value; }); } }
+    return $gullet->readingFromMouth($tokens, sub {
+      $self->read($gullet); }); } }
```

Comment:
Just checking, is `$self->read` "more correct" than the previously hardcoded reading logic?